test_conformance/images/kernel_read_write/test_common.cpp

```diff
@@ -34,122 +34,210 @@ cl_sampler create_sampler(cl_context context, image_sampler_data *sdata, bool te
     return sampler;
 }
 
-void InitFloatCoordsCommon(image_descriptor *imageInfo,
-                           image_sampler_data *imageSampler, float *xOffsets,
-                           float *yOffsets, float *zOffsets, float xfract,
-                           float yfract, float zfract, int normalized_coords,
-                           MTdata d, int lod)
+cl_int get_image_dimensions(image_descriptor *imageInfo, size_t &width,
+                            size_t &height, size_t &depth)
+{
+    cl_int error = CL_SUCCESS;
+    width = imageInfo->width;
+    height = 1;
+    depth = 1;
+    switch (imageInfo->type)
+    {
+        case CL_MEM_OBJECT_IMAGE1D: break;
+        case CL_MEM_OBJECT_IMAGE1D_ARRAY: height = imageInfo->arraySize; break;
+        case CL_MEM_OBJECT_IMAGE2D: height = imageInfo->height; break;
+        case CL_MEM_OBJECT_IMAGE2D_ARRAY:
+            height = imageInfo->height;
+            depth = imageInfo->arraySize;
+            break;
+        case CL_MEM_OBJECT_IMAGE3D:
+            height = imageInfo->height;
+            depth = imageInfo->depth;
+            break;
+        default:
+            log_error("ERROR: Test does not support image type");
+            return TEST_FAIL;
+    }
+    return error;
+}
+
+cl_int InitFloatCoordsCommon(image_descriptor *imageInfo,
+                             image_sampler_data *imageSampler, float *xOffsets,
+                             float *yOffsets, float *zOffsets, float xfract,
+                             float yfract, float zfract, int normalized_coords,
+                             MTdata d, int lod)
 {
     size_t i = 0;
-    if (gDisableOffsets)
+    size_t width_loop, height_loop, depth_loop;
+    cl_int error =
+        get_image_dimensions(imageInfo, width_loop, height_loop, depth_loop);
+    if (!error)
```

Comment:
If _get_image_dimensions_ returns a _test_status_, we shouldn't assume anything about whether _TEST_PASS_ or _TEST_FAIL_ is zero.

Reply:
Hopefully this is resolved by changing to bool rather than test_status.

Reply:
Looks good to me.